common/beerocks/bwl/shared/netlink_socket.cpp

```
@@ -0,0 +1,128 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2020 MaxLinear
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "netlink_socket.h"
+
+#include <netlink/msg.h>
+#include <netlink/netlink.h>
+
+#include <easylogging++.h>
+
+namespace bwl {
+
+netlink_socket::netlink_socket(int protocol)
+    : m_nl_socket(nl_socket_alloc(), nl_socket_free), m_protocol(protocol)
+{
+    LOG_IF(!m_nl_socket, ERROR) << "Failed to allocate netlink socket!";
+}
+
+netlink_socket::~netlink_socket() {}
+
+bool netlink_socket::connect()
+{
+    // Connect the socket
+    if (nl_connect(m_nl_socket.get(), m_protocol) != 0) {
+        LOG(ERROR) << "Failed to connect netlink socket!";
+        return false;
+    }
+
+    return true;
+}
+
+void netlink_socket::close()
+{
+    // Terminate connection and close socket
+    nl_close(m_nl_socket.get());
+}
+
+bool netlink_socket::send_receive_msg(std::function<bool(struct nl_msg *msg)> msg_create,
+                                      std::function<bool(struct nl_msg *msg)> msg_handle)
+{
+    // The Netlink message to send
+    std::unique_ptr<nl_msg, void (*)(nl_msg *)> nl_message(nlmsg_alloc(), nlmsg_free);
+    LOG_IF(!nl_message, FATAL) << "Failed creating netlink message!";
```

So, just to be clear: if allocation fails, we abort here instead of returning false. Except if we remove the AbortOnFatalLog flag, in which case we will log an error and continue as if the allocation succeeded.

The second option is obviously not acceptable. So, either we say that we keep the abort flag and start relying on it to simply crash in out-of-memory conditions (which IMO is very reasonable), or we need to return false here in case allocation failed (but then it should probably not be fatal).

---

#936

---

Changed to log error and return false in that cases